### PR TITLE
"Invalid Tan" message is displayed when the TAN input screen is opened (EXPOSUREAPP-3646)

### PR DESCRIPTION
**Pull request by Oliver-Zimmerman**
### Description

As soon as the user opens the screen for teletan input, a message appears stating that the TAN is invalid. This message remains until the complete TAN has been entered.

**Expected**: Only if the TAN is invalid the message should appear

**Fix**: Added check to only do valid tan check if the length of the TAN is at the max length. Changed unit tests to reflect this. 

### Steps to reproduce

Go to TAN input screen, enter an invalid TAN - the error will now only appear once the user has filled in all the spaces.

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/tan/Tan.kt [hunk 1]
```diff
@@ -10,7 +10,7 @@ data class Tan(
 
     val areCharactersValid = allCharactersValid(value)
     val isTanValidFormat = value.length == MAX_LENGTH && isChecksumValid(value)
-    val isTanValid = areCharactersValid && isTanValidFormat
+    val isTanValid = if (value.length != MAX_LENGTH) true else areCharactersValid && isTanValidFormat
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/tan/Tan.kt:13]: please add some tests

**Review by BMItr — DISMISSED** [review]
code lgtm. tested on Pixel 2 device

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/tan/Tan.kt (hunk 1 above)
Comment (ralfgehrer) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/tan/Tan.kt:13]: Also, shouldn't it rather be (value.length < MAX_LENGTH)

**Review by ralfgehrer — COMMENTED** [review]
Works well on Pixel (Android 10)

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/tan/Tan.kt (hunk 1 above)
Comment (Kolpa) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/tan/Tan.kt:13]: just from a readabilty point of view id prefer if the check for no input is performed in a different place, isTanValid isn't reall true if there is no tan at all.

**Review by ralfgehrer — DISMISSED** [review]

**Review by Kolpa — CHANGES_REQUESTED** [review]
just from a readabilty point of view id prefer if the check for no input is performed in a different place, isTanValid isn't reall true if there is no tan at all.

**Comment by Oliver-Zimmerman** [comment]
> just from a readabilty point of view id prefer if the check for no input is performed in a different place, isTanValid isn't reall true if there is no tan at all.

@Kolpa 
It's not necessarily a check for no input, the issue was that the tan was always invalid when it wasn't at full length so the error message was always there. This addition was just to make sure that the tan validation check is only done if the tan is at full length (complete)

**Comment by Kolpa** [comment]
Just to make my point clear :D i totally agree with the solution and i like it! But the property is called isTanValid, which is quite simply, not true for a tan that is shorter than max length.

So i propose that you do the length check in a different spot and use it in the formatter to decide to not show the error even if the tan is technically "incorrect".

I think doing that way would help a lot with readability for further people looking at this code. Since a check for is tan valid should not return true when a tan is actually invalid but we don't want to display the error.

**Comment by Oliver-Zimmerman** [comment]
@Kolpa 
Noted and implemented. Sorry I missed the initial point of your comment. I have separated the length and validity logic so that it reads better.

**Review by kolyaopahle — APPROVED** [review]
Perfect, thats exactly what i wanted LGTM 👍

**Review by BMItr — APPROVED** [review]

**Pull request merged**